dali/pipeline/data/types.h

```diff
@@ -136,6 +136,9 @@ struct id2type_helper;
 
 /**
  * @brief Compile-time mapping from a type to DALIDataType
+ *
+ * @note If your compiler complains, that "Use of class template `type2id`
```

Comment:
Seems so

Reply:
Basically, I've spent 2+h on figuring out why DALI_TYPE_SWITCH works and TYPE_SWITCH doesn't. The error wasn't descriptive at all. So I've put a comment here. Would you like to propose better solution?

Reply:
I agree - it _technically_ has nothing to do with type2id, but the compiler error is totally baffling when you don't include static_switch - and points you to type2id (at least in case of type_switch), so there's some merit to this message.